src/mode/modeHandler.ts

```diff
@@ -1446,7 +1448,7 @@ export class ModeHandler implements vscode.Disposable {
             await this.handleMultipleKeyEvents(keyStrokes);
           }
 
-          vimState.isReplayingMacro = false;
+          vimState.isReplayingMacro --;
```

Comment:
Doing this, isn't the count 0 at the end of every iteration? So effectively isReplayingMacro will always be either 0 or 1?